src/parquet/util/test-common.h

```diff
@@ -183,6 +183,15 @@ void random_byte_array(int n, uint32_t seed, uint8_t *buf,
     buf += out[i].len;
   }
 }
+
+void random_levels(int n, uint32_t seed, int16_t max_level, int16_t* out) {
+  std::mt19937 gen(seed);
+  std::uniform_int_distribution<uint16_t> d(0, max_level);
+  for (int i = 0; i < n; ++i) {
+    out[i] = d(gen);
+  }
+}
```

Comment:
Perhaps we can refactor `random_numbers` to use another function that accepts upper and lower bounds for the type `T`? Looking again at the data generation in `test-common.h` the code duplication here shows that things are factored suboptimally:

```c++
template <>
void random_numbers(int n, uint32_t seed, int32_t* out) {
  std::mt19937 gen(seed);
  std::uniform_int_distribution<int32_t> d(std::numeric_limits<int32_t>::lowest(),
      std::numeric_limits<int32_t>::max());
  for (int i = 0; i < n; ++i) {
    out[i] = d(gen);
  }
}

template <>
void random_numbers(int n, uint32_t seed, int64_t* out) {
  std::mt19937 gen(seed);
  std::uniform_int_distribution<int64_t> d(std::numeric_limits<int64_t>::lowest(),
      std::numeric_limits<int64_t>::max());
  for (int i = 0; i < n; ++i) {
    out[i] = d(gen);
  }
}
```

I propose:

```c++
template <typename T>
void random_int(int n, uint32_t seed, T lower_bound, T upper_bound, T* out) {
  std::mt19937 gen(seed);
  std::uniform_int_distribution<T> d(lower_bound, upper_bound);
  for (int i = 0; i < n; ++i) {
    out[i] = d(gen);
  }
}
```

Then you can use this for level generation, too.